components/eventing-controller/reconciler/subscription/reconciler.go

```diff
@@ -0,0 +1,964 @@
+		if len(subscription.Status.APIRuleName) != 0 {
+			key := k8stypes.NamespacedName{Namespace: subscription.Namespace, Name: subscription.Status.APIRuleName}
+			previousApiRule := &apigatewayv1alpha1.APIRule{}
+
+			if err := r.Client.Get(ctx, key, previousApiRule); err != nil {
+				if !k8serrors.IsNotFound(err) {
+					return nil, err
+				}
+			} else {
+				// previous ApiRule still exists
+
+				// build a new OwnerReference list without the subscription
+				ownerReferences := make([]v1.OwnerReference, 0, len(previousApiRule.OwnerReferences))
+				for _, ownerReference := range previousApiRule.OwnerReferences {
+					if ownerReference.UID != subscription.UID {
+						ownerReferences = append(ownerReferences, ownerReference)
+					}
+				}
+
+				// delete the previous ApiRule only if the new OwnerReference list is empty
+				if len(ownerReferences) == 0 {
+					if err := r.Client.Delete(ctx, previousApiRule); err != nil {
+						return nil, err
+					}
+				} else if len(previousApiRule.OwnerReferences) > len(ownerReferences) {
+					// update the previous ApiRule only if the new OwnerReference list length is decreased
+					previousApiRule.OwnerReferences = ownerReferences
+					if err := r.Client.Update(ctx, previousApiRule); err != nil {
+						return nil, err
+					}
+				}
+			}
+		}
+
+		err = r.Client.Create(ctx, desiredAPIRule, &client.CreateOptions{})
+		if err != nil {
+			r.eventWarn(subscription, reasonCreateFailed, "Create APIRule failed %s", desiredAPIRule.Name)
+			return nil, errors.Wrap(err, "failed to create APIRule")
+		}
+		r.eventNormal(subscription, reasonCreate, "Created APIRule %s", desiredAPIRule.Name)
+		return desiredAPIRule, nil
+	}
+	logger.Info("Existing APIRules", fmt.Sprintf("in ns: %s for svc: %s", svcNs, svcName), fmt.Sprintf("%s", existingAPIRule.Name))
+
+	object.ApplyExistingAPIRuleAttributes(existingAPIRule, desiredAPIRule)
+	if object.Semantic.DeepEqual(existingAPIRule, desiredAPIRule) {
+		return existingAPIRule, nil
+	}
+	err = r.Client.Update(ctx, desiredAPIRule, &client.UpdateOptions{})
+	if err != nil {
+		r.eventWarn(subscription, reasonUpdateFailed, "Update APIRule failed %s", desiredAPIRule.Name)
+		return nil, errors.Wrap(err, "failed to update an APIRule")
+	}
+	r.eventNormal(subscription, reasonUpdate, "Updated APIRule %s", desiredAPIRule.Name)
+
+	freshExistingAPIRules, err := r.getAPIRulesForASvc(ctx, labels, svcNs)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error while fetching oldApiRules for labels: %v after create/update", labels)
+	}
+	// Cleanup does the following:
+	// 1. Delete APIRule using obsolete ports
+	// 2. Update APIRule by deleting the OwnerReference of the Subscription with port different than that of the APIRule
+	err = r.cleanup(subscription, ctx, subscriptions, freshExistingAPIRules)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to cleanup APIRules")
+	}
+	return desiredAPIRule, nil
```

Comment:
Is this code still required ? 🤔 

My proposal:
- as long as the current tests are green without the above code, we don't need it
- until the remaining test specs are implemented by this [ticket](https://github.com/kyma-project/kyma/issues/9950), we comment it. If these tests show that it is still required, we uncomment it. If not, we delete it in that ticket.

What is your opinion?

Reply:
Removed.